core/src/telemetry/google_cloud.ts

```diff
@@ -0,0 +1,144 @@
+
+import {OTelHooks} from './setup.js';
+
+/**
+ * Configuration options for GCP exporters.
+ */
+export interface GcpExportersConfig {
+  enableCloudTracing?: boolean;
+  enableCloudMetrics?: boolean;
+  enableCloudLogging?: boolean;
+}
+
+/**
+ * Returns GCP OTel exporters to be used in the app.
+ *
+ * @experimental (Experimental, subject to change)
+ * 
+ * @param config Configuration options for enabling different GCP services
+ * @returns OTelHooks containing the configured exporters
+ */
+export async function getGcpExporters(config: GcpExportersConfig = {}): Promise<OTelHooks> {
+  const {
+    enableCloudTracing = false,
+    enableCloudMetrics = false,
+    enableCloudLogging = false,
+  } = config;
+
+  let projectId: string | null = null;
+  
+  try {
+    const auth = new GoogleAuth();
+    projectId = await auth.getProjectId();
+  } catch (error) {
+    logger.warn(
+      'Cannot determine GCP Project. OTel GCP Exporters cannot be set up.' +
+      ' Please make sure to log into correct GCP Project.',
+      error
+    );
+    return {};
+  }
+
+  if (!projectId) {
+    logger.warn(
+      'Cannot determine GCP Project. OTel GCP Exporters cannot be set up.' +
+      ' Please make sure to log into correct GCP Project.'
+    );
+    return {};
+  }
+
+  const spanProcessors: SpanProcessor[] = [];
+  if (enableCloudTracing) {
+    const exporter = await getGcpSpanExporter(projectId);
+    if (exporter) {
+      spanProcessors.push(exporter);
+    }
+  }
+
+  const metricReaders: MetricReader[] = [];
+  if (enableCloudMetrics) {
+    const exporter = await getGcpMetricsExporter(projectId);
+    if (exporter) {
+      metricReaders.push(exporter);
+    }
+  }
+
+  const logRecordProcessors: LogRecordProcessor[] = [];
+  if (enableCloudLogging) {
+    // TODO: Implement GCP Logging exporter
+    // 'Cloud Logging integration is not currently supported. ' +
+    // 'There is no official OpenTelemetry logs exporter for Google Cloud Logging in Node.js. '
+  }
+
+  return {
+    spanProcessors,
+    metricReaders,
+    logRecordProcessors,
+  };
+}
+
+async function getGcpSpanExporter(projectId: string): Promise<SpanProcessor | null> {
+  try {
+    // node seems can not use OTLPTraceExporter to certificate
+    const exporter = new TraceExporter({
+      projectId,
+    });
+    
+    return new BatchSpanProcessor(exporter);
+  } catch (error) {
+    logger.warn('Failed to create GCP trace exporter:', error);
+    return null;
+  }
+}
+
+async function getGcpMetricsExporter(projectId: string): Promise<MetricReader | null> {
+  try {    
+    const exporter = new MetricExporter({
+      projectId,
+    });
+    
+    return new PeriodicExportingMetricReader({
+      exporter,
+      exportIntervalMillis: 5000,
+    });
+  } catch (error) {
+    logger.warn('Failed to create GCP metrics exporter:', error);
+    return null;
+  }
+}
+
+async function getGcpLogsExporter(): Promise<LogRecordProcessor | null> {
```

Comment:
1. not async anymore
2. I don't think we even need this (see my suggested code above)